### PR TITLE
[Addon]Feat/istio addons and doc

### PR DESCRIPTION
**Pull request by hairongGao**
### Description of your changes
1.Fix addon istio installs
2.Add demo for this addon and traits
3.Add istio-vs cue definition

### Checklist

I have:

- [x] Title of the PR starts with type (e.g. `[Addon]` or `[Trait]`).
- [x] Updated/Added any relevant [documentation] and [examples].
- [ ] Unit/E2E Tests added.

### CODE REVIEW

File: experimental/addons/istio/examples/demo.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          # use expose trait to create a svc
+        - type: expose
+          properties:
+            port:
+              - 80
+        - type: istio-gateway
+          properties:
+            hosts:
+              - awesomesite.com
+            # Default value is ingressgateway which is the default selector for your istio svc
```
Comment (wangyikewxgm) [inline experimental/addons/istio/examples/demo.yaml:28]: IMO example files should move from the addon dir to root dir. `/example/istio/....`

**Comment by wangyikewxgm** [comment]
Please don't merge. The trait definition maybe have some issues, I will test it.

File: experimental/addons/istio/resources/istio-vs.cue [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	}
+	description: "Handle Traffic Via Istio VirtualService."
+	labels: {}
+	type: "trait"
+}
+
+template: {
+	output: {
+		apiVersion: "networking.istio.io/v1alpha3"
+		kind:       "VirtualService"
```
Comment (wangyikewxgm) [inline experimental/addons/istio/resources/istio-vs.cue:15]: trait definition is outputs. eg:
```cue
outputs: virtualService {
....
}
```

File: experimental/addons/istio/resources/istio-vs.cue [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	attributes: {
+		appliesToWorkloads: ["*"]
+		podDisruptive: false
+	}
+	description: "Handle Traffic Via Istio VirtualService."
+	labels: {}
+	type: "trait"
+}
+
+template: {
```
Comment (wangyikewxgm) [inline experimental/addons/istio/resources/istio-vs.cue:12]: Should we add destination rule in this trait？

File: experimental/addons/istio/examples/demo.yaml (hunk 1 above)
Comment (hairongGao) [reply experimental/addons/istio/examples/demo.yaml]: commit:
44426672ccaf2098120d58c40c47ea42a344df71
moved to root directory

File: experimental/addons/istio/resources/istio-vs.cue (hunk 2 above)
Comment (hairongGao) [reply experimental/addons/istio/resources/istio-vs.cue]: Good catch , thank you !
Now vela can successfully render resources.But another issue still exist is that configmap schema can't be generated.

File: experimental/addons/istio/resources/istio-vs.cue (hunk 3 above)
Comment (hairongGao) [reply experimental/addons/istio/resources/istio-vs.cue:12]: IMO this cue definition is a pure istio virtualservice, I can add another cue def for destination rules of course .

**Comment by wangyikewxgm** [comment]
Since the experimental addon wouldn't enable by default please enable then disable it in test link [argocd](https://github.com/kubevela/catalog/blob/f362780b41e5c8b72e39333baa260c29cf4fe2ce/test/e2e-test/hack/addon-vela-test.sh#L36)

File: experimental/addons/istio/resources/istio-vs.cue (hunk 3 above)
Comment (wangyikewxgm) [reply experimental/addons/istio/resources/istio-vs.cue:12]: Ok, feel free to add dr trait in next pr.

**Comment by wangyikewxgm** [comment]
fix https://github.com/kubevela/kubevela/issues/3870

**Review by wonderflow — APPROVED** [review]
great work!

**Pull request merged**